runtime/drivers/duckdb/connectors.go

```diff
@@ -19,23 +20,44 @@ func (c *connection) Ingest(ctx context.Context, env *connectors.Env, source *co
 		return err
 	}
 
-	// Driver-specific overrides
-	// switch source.Connector {
-	// case "local_file":
-	// 	return c.ingestFile(ctx, env, source)
-	// }
-
+	// todo :: check if this exceptional handling can be merged
 	if source.Connector == "local_file" {
-		return c.ingestFile(ctx, env, source)
+		conf, err := localfile.ParseConfig(source.Properties)
+		if err != nil {
+			return err
+		}
+		// locally uploaded file
+		if !fileutil.IsGlob(conf.Path) {
+			return c.ingestFile(ctx, env, source)
+		}
+		return c.ingestLocalGlob(ctx, source, conf.Path)
 	}
 
-	path, err := connectors.ConsumeAsFile(ctx, env, source)
+	localPaths, err := connectors.ConsumeAsFile(ctx, env, source)
 	if err != nil {
 		return err
 	}
-	defer os.Remove(path)
+	defer os.RemoveAll(filepath.Dir(localPaths[0]))
+	c.logger.Info(fmt.Sprintf("ingesting files %v", localPaths))
+	// multiple parquet files can be loaded in single sql
+	// this seems to be performing very fast as compared to appending individual files
+	return c.ingestMulti(ctx, source, localPaths)
+}
 
-	return c.ingestFromRawFile(ctx, source, path)
+func (c *connection) ingestLocalGlob(ctx context.Context, source *connectors.Source, glob string) error {
+	query := fmt.Sprintf("CREATE OR REPLACE TABLE %s AS (SELECT * FROM '%s');", source.Name, glob)
+	c.logger.Info("running query %v", zap.String("query", query))
+	return c.Exec(ctx, &drivers.Statement{Query: query, Priority: 1})
+}
```

Comment:
ingestFile has lot of path resolution logic. Also the query pattern for glob ingestion and file ingestion is lot different. I would prefer to keep both functions different. We can rename ingestFile to ingestLocalFile. 
Please let me know your thoughts.

Reply:
For local files, I think we also need the logic in `ingestFile` for globs. Namely, that paths should be resolved relative to `env.RepoDSN` and that we should support custom delimiters for CSVs. So that's why I think it would be best to make it one function.